src/MathExpression/Token.java

```diff
@@ -0,0 +1,39 @@
+package MathExpression;
+
+public class Token {
+    // in order of precedence
+    public enum Type {
+        Number,
+        Parameter,
+        Exponentiation,
+        Division,
+        Multiplication,
+        Addition,
+        Subtraction,
+        Function,
+        Parenthesis_Left,
+        Parenthesis_Right;
+
+        public boolean isOperator() {
```

Comment:
same here could be written as 4-liner. Is it needed? Not really but i like it 👍 
```suggestion
        public boolean isOperator() {
            return switch (this) {
                case Exponentiation, Division, Multiplication, Addition, Subtraction, Function -> true;
                default -> false;
            };
        }
```

Reply:
Didn't know about this notation, here it makes for a lot cleaner code. Done